airflow/timetables/base.py

```diff
@@ -200,3 +203,15 @@ def next_dagrun_info(
             a DagRunInfo object when asked at another time.
         """
         raise NotImplementedError()
+
+    def generate_run_id(
+        self,
+        *,
+        run_type: "DagRunType",
+        logical_date: DateTime,
+        data_interval: Optional[DataInterval],
+        **extra,
+    ) -> str:
+        from airflow.models.dagrun import DagRun
+
+        return DagRun.generate_run_id(run_type, logical_date)
```

Comment:
At first glance, I found it unexpected that the responsibility has moved here, but the code still lives on `DagRun`. Not a request for change, just an observation.

Reply:
Same, this should probably become a free function.

Reply:
Yeah fair comment.

Reply:
I'll move this in to DagRunType (enum) class that we already have.